doc/MANUAL.txt

```diff
@@ -226,17 +268,36 @@ class (the last . splits the class and test method)::
     group_regex=([^\.]+\.)+
 
 
-Remote or isolated test environments
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Profiles and remote or isolated test environments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-A common problem with parallel test running is test runners that use global
-resources such as well known ports, well known database names or predictable
-directories on disk. 
+A common testing problem is testing the same code against a matrix of
+test scenarios. For instance with a C program you may wish to test against
+both `clang` and `gcc`. In Python you may wish to test against multiple
+releases of cPython, or against different Python implementations, or both.
+
+Testr delegates the creation of such a test matrix to you. When test
+identifiers are supplied to testr via listing files, they may be tagged with a
+particular test profile. If they are not tagged, they will be applied to all
+profiles.
+
+Another common issue occurs with with parallel test running where test runners
+that use global resources such as well known ports, well known database names
+or predictable directories on disk. 
 
 One way to solve this is to setup isolated environments such as chroots,
 containers or even separate machines. Such environments typically require
 some coordination when being used to run tests, so testr provides an explicit
-model for working with them.
+model for instantiating and disposing of such test environments. Testr does
+not differentiate between running commands in a regular subprocess, or via ssh
+to an arbitrary endpoint: they are all just test environments. Testr
+will request a separate test environment for each *profile* when
+environments are in use, allowing you to customise the context tests
+run within. If all your environments are homogenous then testr will
+currently overprovision environments when used with profiles. This can
```

Comment:
"If all your environments are homogenous then testr will currently overprovision environments when used with profiles."

I don't know what this means. Perhaps consider expanding this, or using simpler language?